CharlieBackend.Business/Services/StudentService.cs

```diff
@@ -187,5 +187,27 @@ public async Task<StudentDto> GetStudentByEmailAsync(string email)
 
             return _mapper.Map<StudentDto>(student);
         }
+
+        public async Task<Result<StudentDto>> DisableStudentAsync(long id)
+        {
+            var accountId = await GetAccountId(id);
+
+            if (accountId == null)
+            {
+                return Result<StudentDto>.GetError(ErrorCode.NotFound, "Unknown student id.");
+            }
+
+            var student = await GetStudentByAccountIdAsync((long)accountId);
+            var isActive = await _accountService.IsAccountActiveAsync(student.Email);
+            
+            if ((bool)!isActive)
+            {
+                return Result<StudentDto>.GetError(ErrorCode.NotFound, "This account is already disabled.");
+            }
+
+            var disabled = await _accountService.DisableAccountAsync((long)accountId);
```

Comment:
it's uneccessary to create "disabled" variable. You won't use it

Reply:
it will be changed in next pr, check fix all services